cloud_function/replay_from_bigquery.py

```diff
@@ -0,0 +1,104 @@
+import os
+from datetime import datetime, timedelta
+from typing import TypedDict, List
+from zoneinfo import ZoneInfo
+
+from google.cloud import bigquery, pubsub, scheduler
+from google.protobuf import json_format
+
+QUERY = "SELECT * FROM {} WHERE publish_time > @start AND publish_time < @end"
+CURRENT_TIMEZONE = ZoneInfo("America/Los_Angeles")
+
+
+class DataSourceConfig(TypedDict):
+    table_name: str  # big query table to read data from
+    proto_package: str  # package containing protobuf definitions
+    proto_class: str  # class of the protobuf message
+    topic_id: str  # topic ID for the replayed data
+    fetch_interval: int  # duration, in minutes, before the simulation's current time for range queries.
+
+
+class TimeConfig:
```

Comment:
You want to use a dataclass here -or one of the alternatives, i.e. the attr library or a Pydantic model

Reply:
added! (not sure why it doesn't show up here; but if you go to "Files Changed" it will ship up)